fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraWebACUserSecurityContext.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.auth.common;
+
+import java.security.Principal;
+
+import org.apache.commons.lang.StringUtils;
+import org.modeshape.jcr.security.SecurityContext;
+import org.modeshape.jcr.value.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The security context for Fedora WebAC servlet users. These users are not
+ * necessarily authenticated by the container, i.e. users may include the
+ * general public. This security context delegates all access decisions to the
+ * configured authorization delegate.
+ *
+ * @author mohideen
+ */
+public class FedoraWebACUserSecurityContext extends FedoraUserSecurityContext {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(FedoraWebACUserSecurityContext.class);
+
+    /**
+     * Constructs a new security context.
+     *
+     * @param userPrincipal the user principal associated with this security
+     *        context
+     * @param fad the authorization delegate
+     */
+    protected FedoraWebACUserSecurityContext(final Principal userPrincipal,
+                                        final FedoraAuthorizationDelegate fad) {
+        super(userPrincipal, fad);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see SecurityContext#hasRole(String)
+     */
+    @Override
+    public final boolean hasRole(final String roleName) {
+        // Under this custom PEP regime, all users have modeshape read and write
+        // roles.
+        if ("http://www.w3.org/ns/auth/acl#Read".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Write".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Append".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Control".equals(roleName)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Get the user principal associated with this context.
+     *
+     * @return the user principal associated with this security context
+     */
+    public Principal getEffectiveUserPrincipal() {
+        if (this.loggedIn && this.userPrincipal != null) {
+            return this.userPrincipal;
+        }
+        return fad.getEveryonePrincipal();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see
+     * org.modeshape.jcr.security.AdvancedAuthorizationProvider#hasPermission
+     * (org.modeshape.jcr.security.AdvancedAuthorizationProvider.Context,
+     * org.modeshape.jcr.value.Path, java.lang.String[])
+     */
+    @Override
+    public boolean hasPermission(final Context context, final Path absPath,
+                                 final String... actions) {
+
+        LOGGER.debug("Verifying hasPermission on path: " + absPath + " for: " + StringUtils.join(actions));
```

Comment:
Please use the log4j syntax:
`LOGGER.debug("Verifying hasPermission on path: {} for: {}", absPath, StringUtils.join(actions));`